internal/configs/version2/nginx-plus.virtualserver.tmpl

```diff
@@ -1,6 +1,6 @@
 {{ range $u := .Upstreams }}
 upstream {{ $u.Name }} {
-    zone {{ $u.Name }} 256k;
+    zone {{ $u.Name }} {{ if ne $u.UpstreamZoneSize "0" }}{{ $u.UpstreamZoneSize }}; {{ else }} 256k;{{ end }}
```

Comment:
this wasn't simplified as suggested in https://github.com/nginxinc/kubernetes-ingress/pull/659#discussion_r314906070